mod/directory.php

```diff
@@ -93,8 +97,17 @@ function directory_content(&$a) {
 
 		foreach($r as $rr) {
 
+			$community = '';
 
-			$profile_link = $a->get_baseurl() . '/profile/' . ((strlen($rr['nickname'])) ? $rr['nickname'] : $rr['profile_uid']);
+			// get the friendica address and the profile url of the user
+			$p = q("SELECT `addr` AS faddr, `url` AS profile_url FROM `contact` WHERE `uid` = %d AND `self`",
+				intval($rr['uid'])
```

Comment:
This query should be done in the query part before the loop. This reduces the amount of queries.